### PR TITLE
Nerf acid damage against materials

### PR DESCRIPTION
**Pull request by PropaneSoup**
After playing around with spitter zombies, I remember some requests for an acid nerf, so I went through the materials json and upped a lot of the acid resist numbers, hoping to bring some sense to it. For example, glass and superalloy had an acid resist of 0? A neat side effect of this: walking through acid with rubber boots will damage you much less than naked!

Feel free to comment on whether some values should be up or down, but I feel the most of these are sensible.

### CODE REVIEW

File: data/json/materials.json [hunk 1]
```diff
@@ -31,7 +31,7 @@
             "very damaged",
             "thoroughly damaged"
         ],
-        "acid_resist" : 1,
```
Comment (Frost-wood) [inline data/json/materials.json]: Human flesh, veggy etc should remain at 1

File: data/json/materials.json [hunk 2]
```diff
@@ -453,7 +453,7 @@
             "cracked",
             "shattered"
         ],
-        "acid_resist" : 3,
+        "acid_resist" : 9,
```
Comment (Frost-wood) [inline data/json/materials.json]: Glass resistance I argue should be even higher.

**Comment by Coolthulhu** [comment]
Personally I'd rather have someone change the acid mechanics (current 1d10 roll doesn't properly handle it), but this still looks good.

**Comment by Coolthulhu** [comment]
Needs balancing.
Leather and steel are 100% immune.
Plastic, from which all the hazmat gear is made, is resistant but much less so than leather.

You may want to change both the acid function in `field.cpp` and the acid resistance function in `item.cpp`. The former has a hard limit on acid resistance (anything above 8 is 100% immune), the latter multiplies all acid resistance by 1.5.
This means that anything with acid resistance of 6 or more is totally immune.

The way it should work:
* Steel should not be immune. It can take long time to rust compared to organics (no hard cap on resistance), but it shouldn't just shrug the acid forever.
* Superalloy and hardsteel probably aren't made to resist corrosion, but mechanical damage. They aren't sci-fi adamantium.
* Hazmat gear like rubber shoes, hazmat suit etc. should be very resistant, possibly immune. You could either buff all plastics to hazmat quality or add a new chemically resistant plastic material. Plastic could also be special cased to resist all acid (same for glass, ceramics and gold) like nomex is for fire.


You can test it by debug-mutating Debug invincibility (not Invisibility) and spawning a big horde of spitters and survivor zeds. Spitter acid will kill the survivor zeds and make them drop all sorts of high-level loot. You'll then easily notice that some materials stay in the acid pile forever, while others melt quickly.

**Comment by Snaaty** [comment]
Hm, plastic being immune is kind a tricky I imagine, because for example a plastic dog whistle should not be immune against acid.

**Comment by PropaneSoup** [comment]
Yea, leather was way too high. I lowered it, and a few of the metals down to reasonable levels, I also upped plastic to 9 and glass, ceramics and a few of the precious metals to 10. With this, I also removed the * 1.5 that all resistances were getting. So now plastic is nigh-immune, and anything 10 or above *is* immune.

This plays much better now, spitters aren't just a game of trying to keep them away from your stuff anymore, and dressing in acid-resistant clothing actually helps a lot now.

**Comment by Coolthulhu** [comment]
Does the armor acid resistance help against acid damage to player?
If so, it would probably need to be rescaled too. The damage to a naked survivor and one in leather boots shouldn't differ too much (50% or so at most).

I'm not sure what should be done with kevlar. It is supposed to be quite resistant to acids IRL, but it probably isn't very good at protecting the wearer from them.

**Comment by PropaneSoup** [comment]
> Does the armor acid resistance help against acid damage to player?

As far as I can tell it does. Starting at line 80 in damage.ccp. I'm not 100% sure how it deals with things like coverage, ect., but I know wearing something is better than nothing against acid in-game.

**Comment by graysage1** [comment]
@PropaneSoup 

>and dressing in acid-resistant clothing actually helps a lot now.

Is acid resistance currently a secret field in-game that can only be learned about by looking at an online database like http://cdda-trunk.estilofusion.com/ ?  If so, wouldn't it be better to include acid and fire resistance properties into item description the way bash/cut resistance are included already?  That way players can make intelligent decisions without having to go online for help.

**Comment by PropaneSoup** [comment]
> Is acid resistance currently a secret field in-game that can only be learned about by looking at an online database like http://cdda-trunk.estilofusion.com/ ?

Acid resistance is actually calculated based on an item's materials. So everything made of leather is going to have the same acid resistance. It only gets tricky when you have multiple materials. Then it just takes the average of the two materials' acid resistance. Whether or not it's something that would be useful in item descriptions, I don't think so. Acid is a not a really common field in-game since acid rain was removed, and most of these resistances, I think, are common sense.

**Comment by graysage1** [comment]
>Acid is a not a really common field in-game since acid rain was removed, and most of these resistances, I think, are common sense.

I still think that the players can use a little help here.  To me it's not all common sense.  Maybe the help screen '?' can include something about fire/acid resistance at least?

**Comment by PropaneSoup** [comment]
Ok, I added a quick section on acid damage in the survival tips section of the help menu.

**Comment by Coolthulhu** [comment]
Merged now, but something will need to be done about the decreased acid damage against player. Best idea I can think of would be giving acid armor penetration, so that leather doesn't do much against it.

**Pull request merged**